data_steward/notebooks/bq.py

```diff
@@ -7,3 +7,36 @@
 def query(q, use_cache=False):
     query_job_config = bigquery.job.QueryJobConfig(use_query_cache=use_cache)
     return client.query(q, job_config=query_job_config).to_dataframe()
+
+
+def list_datasets():
+    """
+    Lists all datasets existing in a project.
+
+    :return: List of dataset objects
+    """
+    datasets = list(client.list_datasets())
+    return datasets
+
+
+def delete_dataset(project_id,
+                   dataset_id,
+                   delete_contents=True,
+                   not_found_ok=True):
+    """
+    Delete a dataset in a project 
+    
+    :param project_id: Identifies the project the containing the dataset
+    :param dataset_id: Identifies the dataset to delete
+    :param delete_contents: If set True, deletes all contents within the dataset
+    :param not_found_ok: If set True, does not raise error if dataset cannot be found
+    :return: 
+    """
+    if client.project == project_id:
+        client.delete_dataset(dataset_id,
+                              delete_contents=delete_contents,
+                              not_found_ok=not_found_ok)
+    else:
+        raise ValueError(
+            'Wrong project set. Please set the GOOGLE_CLOUD_PROJECT env variable correctly'
+        )
```

Comment:
I like the idea of adding a fail safe, especially because this is a delete function and we can't predict if modules using this function are going to properly implement safety checks.  But I don't know if we want to fall back to environment variables or force setting environment variables.  I think we need to create a utility function to get a client object that requires setting the project (possibly credentials).  You can review the client api here:  https://googleapis.dev/python/bigquery/latest/generated/google.cloud.bigquery.client.Client.html#google.cloud.bigquery.client.Client .

Then that would be the first call you would make here, to get the client (just like in `bq_utils` functions).  The problem with a top level assignment is that it can't be mocked so it really messes up unit testing.  I know you didn't add that, but that's why I'm suggesting creating a function similar to `bq_utils.create_service()`.  But I'd like to start breaking our inherent dependency on environment variable settings.

Reply:
Sure, I will rebase onto develop now that we have it